onvm/onvm_nflib/onvm_flow_table.h

```diff
@@ -148,18 +148,17 @@ onvm_ft_fill_key(struct onvm_ft_ipv4_5tuple *key, struct rte_mbuf *pkt) {
                 return -EPROTONOSUPPORT;
         }
         ipv4_hdr = onvm_pkt_ipv4_hdr(pkt);
-        memset(key, 0, sizeof(struct onvm_ft_ipv4_5tuple));
         key->proto = ipv4_hdr->next_proto_id;
         key->src_addr = ipv4_hdr->src_addr;
         key->dst_addr = ipv4_hdr->dst_addr;
         if (key->proto == IP_PROTOCOL_TCP) {
                 tcp_hdr = onvm_pkt_tcp_hdr(pkt);
-                key->src_port = tcp_hdr->src_port;
-                key->dst_port = tcp_hdr->dst_port;
+                key->src_port = rte_cpu_to_be_16(tcp_hdr->src_port);
+                key->dst_port = rte_cpu_to_be_16(tcp_hdr->dst_port);
```

Comment:
Convert to big endian. DPDK has examples of using these here https://doc.dpdk.org/api/examples_2performance-thread_2l3fwd-thread_2main_8c-example.html.